CodeModel/CodeParsers/CSharp/ICBMCGraph.cs

```diff
@@ -0,0 +1,336 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using CodeModel.CaDETModel.CodeItems;
+
+namespace CodeModel.CodeParsers.CSharp
+{
+    public class ICBMCGraph
+    {
+        private int[,] Matrix { get; }
+        private List<Edge> EdgesInGraph { get; }
+        internal List<SubGraphPair> SubGraphPairs { get; set; }
+
+        public ICBMCGraph(CaDETClass parsedClass)
+        {
+            var normalMethods = parsedClass.Members.Where(IsMemberNormalMethod).ToList();
+            var fields = parsedClass.Fields;
+            var fieldDefiningAccessors =
+                parsedClass.Members.Where(m => m.IsFieldDefiningAccessor()).ToList();
+            Matrix = InitializeMatrix(normalMethods, fields, fieldDefiningAccessors);
+            EdgesInGraph = GetAllEdgesInGraph();
+            SubGraphPairs = GetSubGraphPairs();
+        }
+
+        private ICBMCGraph(int[,] matrix)
+        {
+            Matrix = matrix.Clone() as int[,];
+            EdgesInGraph = GetAllEdgesInGraph();
+            SubGraphPairs = GetSubGraphPairs();
+        }
+
+        /// <summary>
+        /// Member is normal method if it's not constructor, nor getter or setter,
+        /// nor delegate function.
+        /// </summary>
+        private bool IsMemberNormalMethod(CaDETMember m)
+        {
+            return m.Type == CaDETMemberType.Method &&
+                   m.Metrics[CaDETMetric.MELOC] > 1; // && m.Modifiers.Contains(new CaDETModifier("public"));
+        }
+
+        private int[,] InitializeMatrix(List<CaDETMember> normalMethods, List<CaDETField> fields,
+            List<CaDETMember> fieldDefiningAccessors)
+        {
+            int[,] matrix = new int[normalMethods.Count, fields.Count + fieldDefiningAccessors.Count];
+            for (var i = 0; i < normalMethods.Count; i++)
+            {
+                for (var j = 0; j < fields.Count; j++)
```

Comment:
There are 4-5 places where you have to check if we are working with fields or methods, and the code is mostly the same for both cases... This seems like a smell, but I'm not sure how we could refactor this. We might know more once we start transforming this to include indirect field access through private methods etc.